packages/caver-klay/caver-klay-accounts/src/index.js

```diff
@@ -136,10 +146,187 @@ Accounts.prototype._determineAddress = function _determineAddress(legacyAccount,
+ *
+ * @method createAccountKey
+ * @param {String|Array|Object} accountKey Parameters to be used when creating the AccountKey.
+ * @return {Object}
+ */
+Accounts.prototype.createAccountKey = function createAccountKey(accountKey) {
+  if (Account.isAccountKey(accountKey)) accountKey = accountKey.keys
+  
+  if (_.isString(accountKey)) {
+    accountKey = this.createAccountKeyPublic(accountKey)
+  } else if (_.isArray(accountKey)) {
+    accountKey = this.createAccountKeyMultiSig(accountKey)
+  } else if (_.isObject(accountKey)) {
+    accountKey = this.createAccountKeyRoleBased(accountKey)
+  } else {
+    throw new Error(`Invalid accountKey type: ${typeof accountKey}`)
+  }
+  return accountKey
+}
+
+/**
+ * createAccountKeyPublic creates AccountKeyPublic with a string of private key.
+ *
+ * @method createAccountKeyPublic
+ * @param {String} privateKey Private key string that will be used to create AccountKeyPublic.
+ * @return {Object}
+ */
+Accounts.prototype.createAccountKeyPublic = function createAccountKeyPublic(privateKey) { 
+  if (privateKey instanceof AccountKeyPublic) return privateKey
+
+  if (!_.isString(privateKey)) throw new Error('Creating a AccountKeyPublic requires a private key string.')
+  
+  const parsed = utils.parsePrivateKey(privateKey)
+  privateKey = parsed.privateKey
+
+  if (!utils.isValidPrivateKey(privateKey)) throw new Error(`Failed to create AccountKeyPublic. Invalid private key : ${privateKey}`)
+
+  return new AccountKeyPublic(privateKey)
+}
+
+/**
+ * createAccountKeyMultiSig creates AccountKeyMultiSig with an array of private key.
+ *
+ * @method createAccountKeyMultiSig
+ * @param {Array} privateKeys An Array of private key string that will be used to create AccountKeyMultiSig.
+ * @return {Object}
+ */
+Accounts.prototype.createAccountKeyMultiSig = function createAccountKeyMultiSig(privateKeys) {
+  if (privateKeys instanceof AccountKeyMultiSig) return privateKeys
+
+  if (!_.isArray(privateKeys)) throw new Error('Creating a AccountKeyMultiSig requires an array of private key string.')
+
+  for (let p of privateKeys) {
+    const parsed = utils.parsePrivateKey(p)
+    p = parsed.privateKey
+    if (!utils.isValidPrivateKey(p)) throw new Error(`Failed to create AccountKeyMultiSig. Invalid private key : ${p}`)
+  }
+
+  return new AccountKeyMultiSig(privateKeys)
+}
+
+/**
+ * createAccountKeyRoleBased creates AccountKeyRoleBased with an obejct of key.
+ *
+ * @method createAccountKeyRoleBased
+ * @param {Object} keyObject Object that defines key for each role to use when creating AccountKeyRoleBased.
+ * @return {Object}
+ */
+Accounts.prototype.createAccountKeyRoleBased = function createAccountKeyRoleBased(keyObject) {
+  if (keyObject instanceof AccountKeyRoleBased) return keyObject
+
+  if (!_.isObject(keyObject) || _.isArray(keyObject)) throw new Error('Creating a AccountKeyRoleBased requires an object.')
```

Comment:
Is this condition `|| _.isArray(keyObject)` necessary?

Reply:
Yes i think necessary, because In javascript array also can be an Object type.